File: dotnet/Src/ButterflyNet.Parser/Scopes/PreformattedScope.cs
namespace ButterflyNet.Parser.Scopes {
	public class PreformattedScope : BlockScope {
		private readonly string language;

		public PreformattedScope(string language) {
			this.language = language;
		}

		public override void Open(ButterflyAnalyzer analyzer) {
			analyzer.OpenPreformatted(language);
		}

		public override void Close(ButterflyAnalyzer analyzer) {
			analyzer.ClosePreformatted(language);
		}
	}
}

File: src/js/src/scopes/PreformattedScope.js
function PreformattedScope(language) {
	this.language = language;
	this.type = ScopeTypeCache.preformatted;
	this.level = ScopeLevel.block;
	this.canNestText = true;
	
	this.open = function(analyzer) {
		analyzer.openPreformatted(this.language);
	};
	this.close = function(analyzer) {
		analyzer.closePreformatted(this.language);
	};
}

extend(Scope, PreformattedScope);